File: src/main/intellibitz/sted/event/KeypadEvent.java
package intellibitz.sted.event;

import java.util.EventObject;

/**
 * holds the source Thread information
 */
public class KeypadEvent
        extends EventObject
        implements Cloneable {
    private IKeypadEventSource keypadEventSource;

    /**
     * @param src the source for this event
     */
    public KeypadEvent(IKeypadEventSource src) {
        super(src);
        keypadEventSource = src;
    }

    /**
     * @return IKeypadEventSource the Source which generated this Event
     */
    public IKeypadEventSource getEventSource() {
        return keypadEventSource;
    }

    public Object clone()
            throws CloneNotSupportedException {
        return super.clone();
    }
}

File: src/main/intellibitz/sted/event/KeypadEvent.kt
package sted.event

import java.util.*

/**
 * holds the source Thread information
 */
class KeypadEvent(val eventSource: IKeypadEventSource) : EventObject(eventSource), Cloneable {
    @Throws(CloneNotSupportedException::class)
    public override fun clone(): Any {
        return super.clone()
    }
}